api/validation/etcd.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"strings"
+
+	"github.com/gardener/etcd-druid/api/v1alpha1"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateEtcd validates Etcd object.
+func ValidateEtcd(etcd *v1alpha1.Etcd) field.ErrorList {
+	return validateEtcdSpec(etcd, field.NewPath("spec"))
+}
+
+// ValidateEtcdUpdate validates a Etcd object before an update.
+func ValidateEtcdUpdate(new, old *v1alpha1.Etcd) field.ErrorList {
+	return validateEtcdSpecUpdate(&new.Spec, &old.Spec, field.NewPath("spec"))
+}
+
+func validateEtcdSpec(etcd *v1alpha1.Etcd, path *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	prefix := etcd.Spec.Backup.Store.Prefix
+	if !validatePrefixName(prefix, etcd.Name, etcd.Namespace) {
+		allErrs = append(allErrs, field.Invalid(path.Child("backup.store.prefix"), prefix, "field is required"))
```

Comment:
The error message "field is required" is wrong, and it can't be right since at this point you've lost the information what is actually wrong. Instead, you should write this as:
```suggestion
	allErrs = append(allErrs, validateBackupStorePrefix(prefix, etcd.Name, etcd.Namespace, path.Child("backup.store.prefix"))...)
```
And take care to properly report all validation errors via e.g. `field.Invalid` (there might be more than one) in `validateBackupStorePrefix`. This is a general pattern to be followed in all validation code.